src/components/Footer/footer.test.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { shallow, configure } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+import ContactPanels from './ContactPanels/index'
+
+configure({ adapter: new Adapter() })
+
+const renderComponent = (props = {}) => {
+  const component = shallow(<ContactPanels {...props} />)
+  return component
+}
+
+const openingHours = jest
+  .fn()
+  .mockReturnValue(true)
+  .mockReturnValueOnce(false)
+  .mockReturnValueOnce(true)
+
+describe('Footer Component', () => {
+  describe('Contact Panels', () => {
+    let component, disabled
+    beforeEach(() => {
+      component = renderComponent()
+      disabled = component
+        .find('StyledContactPanels__FooterButtons')
+        .first()
+        .props('disabled')
+    })
+    it('ContactPanels Component', () => {
+      expect(component.exists()).toBe(true)
+    })
+
+    describe('Opening Hours', () => {
+      it('Should return true on sundays, saturdays after 3pm and Mondays to Friday after 6pm', () => {
```

Comment:
I think this is not testing what it says it is: looks to me like it's mocking the value for `openingHours` and testing that against the disabled state of the button (though this in itself of course a valid test).

Would it be better to achieve this to set a state on the component for the opening hours and  then test against this?

Disclaimer: I'm very unfamiliar with Jest and happy to go through it in more detail.

Reply:
Well, thats another way of achieving the same result. The function will still be mocked. I would prefer that method but introducing state management to a style library looks like an overkill to me. Regarding the descriptive of the test, we can change that to something more descriptive. @davidtrussler what do you think?

Reply:
> Well, thats another way of achieving the same result. The function will still be mocked. I would prefer that method too but introducing state management to a style library looks like an overkill to me. Regarding the descriptiveness of the test, we can change that to something more descriptive. @davidtrussler what do you think?

OK I think there’s a few issues here and I haven’t had much time yet to look too deeply into this. But I think this library can be seen as a component library rather than purely a style library. With that in mind I think it is fine for this logic to exist here and therefore we could use a stateful component to store the value for opening hours. That would in turn possibly make it easier to run the tests on this component by passing in a value for the date and testing that the returned value is as expected. Usual caveats apply with regard to my own slight knowledge of how Jest works but I shall have a quick look into this ASAP.